activesupport/lib/active_support/core_ext/time/calculations.rb

```diff
@@ -319,7 +319,12 @@ def compare_with_coercion(other)
     if other.class == Time
       compare_without_coercion(other)
     elsif other.is_a?(Time)
-      compare_without_coercion(other.to_time)
+      # also avoid ActiveSupport::TimeWithZone#to_time before Rails 8.0
+      if other.respond_to?(:comparable_time)
+        compare_without_coercion(other.comparable_time)
+      else
+        compare_without_coercion(other.to_time)
+      end
```

Comment:
(non-blocking) If we know this is a Time subclass (or is claiming to be one like TimeWithZone) how would you feel about just using `utc` (since that exists on time and is the same as `comparable_time`)?

```suggestion
      compare_without_coercion(other.utc)
```